app/api/getProducts/route.ts

```diff
@@ -0,0 +1,49 @@
+// app/api/post/route.ts
+import postgres from '@/app/utils/postgres';
+import { NextRequest, NextResponse } from 'next/server';
+
+export async function POST(request: NextRequest) {
+    try {
+        const client = await postgres.connect();
+
+        const { productName, itemCode, quantity: rawQuantity } = await request.json();
+        const quantity = rawQuantity ? parseInt(rawQuantity) : null;
+
+        if (!productName || !itemCode || !quantity || quantity < 0) {
+            return NextResponse.json({ error: 'Please provide all required fields' }, { status: 400 });
+        }
+        const result = await client.query(
```

Comment:
Just a thought: Would it be better to use the drizzle ORM features here instead of raw SQL? Might be open to SQL injection.